workspace/workspace.go

```diff
@@ -0,0 +1,105 @@
+package workspace
+
+import (
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+var rgxSerialSuffix = regexp.MustCompile(`-\d*$`)
+
+// Workspace represents a user's Exercism workspace.
+// It may contain a user's own exercises, and other people's
+// exercises that they've downloaded to look at or run locally.
+type Workspace struct {
+	Dir string
+}
+
+// New returns a configured workspace.
+func New(dir string) Workspace {
+	return Workspace{Dir: dir}
+}
+
+// Locate the matching directories within the workspace.
+// This will look for an exact match on absolute or relative paths.
+// If given the base name of a directory with no path information it
+// It will look for all directories with that name, or that are
+// named with a numerical suffix.
+func (ws Workspace) Locate(exercise string) ([]string, error) {
+	var paths []string
+
+	// First assume it's a path.
+	dir := exercise
+
+	// If it's not an absolute path, make it one.
+	if !filepath.IsAbs(dir) {
+		var err error
+		dir, err = filepath.Abs(dir)
+		if err != nil {
+			return paths, err
+		}
+	}
+
+	// If it exists, we were right. It's a path.
+	if _, err := os.Stat(dir); err == nil {
+		if !strings.HasPrefix(dir, ws.Dir) {
+			return paths, ErrNotInWorkspace
+		}
+
+		src, err := filepath.EvalSymlinks(dir)
```

Comment:
What kind of errors can we expect from this function? Is there anything worth testing against?